### PR TITLE
Anisha/calc

### PR DESCRIPTION
**Pull request by k-anisha**
## 📝 Summary

Adding rates of change notebook for Calculus in ML

## 📋 Checklist

- [ ] I have included package dependencies in the notebook file [using `--sandbox`](https://docs.marimo.io/guides/package_reproducibility/)
- [x] If adding a course, include a `README.md`
- [x] Keep language direct and simple.

### CODE REVIEW

**Comment by Haleshot** [comment]
Relevant issue #58

File: calculus_for_ML/RatesOfChange.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    from sympy import diff
+    return diff, mo, np, plt, symbols
+
+
+@app.cell
+def _(mo):
+    mo.md(
+        r"""
+        In this notebook, we'll learn about rates of change in Machine Learning. 
+
```
Comment (Haleshot) [inline calculus_for_ML/RatesOfChange.py:22]: I'd start the notebook with the title being mentioned (h1 tag) followed by the author's name. Something like

```markdown
# Rates of Change in Machine Learning

*By [YourName](https://github.com/k-anisha)*

In this notebook, we'll learn about...
```

File: calculus_for_ML/RatesOfChange.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+@app.cell
+def _():
+    import marimo as mo
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from sympy import symbols
+    from sympy import diff
+    return diff, mo, np, plt, symbols
```
Comment (Haleshot) [inline calculus_for_ML/RatesOfChange.py:14]: I'd move this cell to the bottom (better to directly start with the notebook topic - also consistent and followed in most notebooks in the learn repo).

File: calculus_for_ML/RatesOfChange.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    def f(x):
+        return x ** 2
+
+    # Now, we can plot the function.abs
+
+    x = np.linspace(-10, 10, 500)
+    y = f(x)
+
+    plt.plot(x, y, label=r'$f(x) = x^2$')
+    plt.title('Graph of f(x) = x^2')
```
Comment (Haleshot) [inline calculus_for_ML/RatesOfChange.py:48]: Instead of a static plot, consider adding sliders to explore different functions:

```python
# Could use mo.ui.slider to let users change the coefficient
coeff = mo.ui.slider(1, 5, value=1, label="Coefficient")
def f(x):
    return coeff.value * x**2
```

This would let users see how changing the coefficient affects the parabola (+ also adds to the interactivity of the notebook).

File: calculus_for_ML/RatesOfChange.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    # Our parameters below
+
+    learning_rate = 0.1
+    initial = 8
+    iterations = 20
+
+    current = initial
+    history = [current]
+
+    for i in range(iterations):
```
Comment (Haleshot) [inline calculus_for_ML/RatesOfChange.py:120]: The learning rate, initial point, & iterations are hardcoded. Consider making these interactive:

```python
learning_rate = mo.ui.slider(0.01, 0.5, value=0.1, step=0.01, label="Learning Rate")
initial = mo.ui.slider(-10, 10, value=8, label="Starting Point") 
iterations = mo.ui.slider(5, 50, value=20, label="Iterations")
```

Users could experiment with different values and see how it affects convergence.

File: calculus_for_ML/RatesOfChange.py [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+        Let's get into rates of change!
+        """
+    )
+    return
+
+
+@app.cell
+def _(np, plt):
+    # First, let's start by definiting a simple function.
```
Comment (Haleshot) [inline calculus_for_ML/RatesOfChange.py:37]: Btw, I'd recommend checking out our plotting tutorial: https://links.marimo.app/tutorial-plotting
Has a good example where various functions are visualized iteratively.

**Review by Haleshot — COMMENTED** [review]
The math content throughout the notebook is solid - the derivatives, the GD implementation & physics relationships you've laid out are accurate. The economics example at the end also makes good sense and ties things together nicely.

For styling, you could polish this up with some markdown formatting. Using backticks around function names like `f(x)` and `f'(x)` would make them stand out better.

Overall, I really like how you build things step-by-step (the flow). Would recommend improving the above PR description to be more descriptive and capture the contents of the notebook accurately (refer to [previous PRs](https://github.com/marimo-team/learn/pulls?q=is%3Apr+is%3Aclosed)).

Also, a nice summary at the end of the notebook going over the contents of the notebook (what was covered, etc.). would be nice. Usage of appropriate [markdown extensions](https://docs.marimo.io/guides/outputs/#markdown-extensions) &/or [admonitions](https://docs.marimo.io/guides/outputs/#admonitions) would also make it stand out better.
